spring-amqp/src/test/java/org/springframework/amqp/core/BindingBuilderTests.java

```diff
@@ -16,38 +16,83 @@
 
 package org.springframework.amqp.core;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
 import java.util.Collections;
 
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 /**
  * @author Mark Fisher
  */
 public class BindingBuilderTests {
 
+	private static Queue queue;
+
+	@BeforeClass
+	public static void setUp() {
+		queue = new Queue("q");
+	}
+
 	@Test
 	public void fanoutBinding() {
-		Binding binding = BindingBuilder.bind(new Queue("q")).to(new FanoutExchange("f"));
+		FanoutExchange fanoutExchange = new FanoutExchange("f");
+		Binding binding = BindingBuilder.bind(queue).to(fanoutExchange);
+		assertEquals(fanoutExchange.getName(), binding.getExchange());
+		assertEquals("", binding.getRoutingKey());
+		assertEquals(Binding.DestinationType.QUEUE, binding.getDestinationType());
+		assertEquals(queue.getName(), binding.getDestination());
 		assertNotNull(binding);
```

Comment:
Thank you for so fast turnaround @neetkee !

Well, this `assertNotNull()`  must be first any way. Otherwise we end up with ``NPE` already on a first `binding.getExchange()`.

Please, fix that together with your name to the `@author` list and update for Copyright to the current year - `2002-2017`.

After that we will merge it.